src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -86,7 +86,7 @@ struct FrequencyAndIndexStruct {
     std::uint32_t const TermFrequency;                            // Words and the number of documents that it appears in
     std::uint32_t const Index;                              
 
-    //will cause error if the struct marked deleted after copy
+    //copy needed in generating FrequencyAndIndexMap
```

Comment:
Perhaps use the map's emplace method? If that doesn't work out, let me know which line has the problems.

Reply:
highly suspect line 407, but not 100% sure as the error message doesn't target any. And it looks emplace() doesn't work out here.

Reply:
Try to change 
```
termFrequencyAndIndex.insert(std::make_pair(termFrequencyIter.first, FrequencyAndIndexStruct(termFrequencyIter.second, termIndex.at(termFrequencyIter.first))));
    }
```
to
```
FrequencyAndIndexStruct value(termFrequencyIter.second, termIndex.at(termFrequencyIter.first));
termFrequenceAndIndex.emplace(std::make_pair(termFrequencyIter.first, std::move(value)));
```

If that doesn't work:
```
FrequencyAndIndexStruct value(termFrequencyIter.second, termIndex.at(termFrequencyIter.first));
termFrequenceAndIndex.emplace(<The Map Type Here>::value_type(termFrequencyIter.first, std::move(value)));
```

Reply:
looks both not work, working on trying other ways...